common/src/main/java/com/sedmelluq/discord/lavaplayer/natives/NativeLibLoader.java

```diff
@@ -20,17 +22,72 @@
  * Loads native libraries by name. Libraries are expected to be in classpath /natives/[arch]/[prefix]name[suffix]
  */
 public class NativeLibLoader {
-  public static final String LINUX_X86 = "linux-x86";
-  public static final String LINUX_X86_64 = "linux-x86-64";
-  public static final String LINUX_ARM = "linux-arm";
-  public static final String LINUX_ARM_64 = "linux-aarch64";
-  public static final String WIN_X86 = "win-x86";
-  public static final String WIN_X86_64 = "win-x86-64";
-  public static final String DARWIN = "darwin";
+  public static final String OS_DARWIN = "darwin";
+  public static final String OS_LINUX = "linux";
+  public static final String OS_OSX = "osx";
+  public static final String OS_SOLARIS = "solaris";
+  public static final String OS_WINDOWS = "win";
+
+  public static final String ARCH_ARM = "arm";
+  public static final String ARCH_ARM_HF = "armhf";
+  public static final String ARCH_ARMv8_32 = "aarch32";
+  public static final String ARCH_ARMv8_64 = "aarch64";
+  public static final String ARCH_MIPS_32 = "mips";
+  public static final String ARCH_MIPS_32_LE = "mipsel";
+  public static final String ARCH_MIPS_64 = "mips64";
+  public static final String ARCH_MIPS_64_LE = "mips64el";
+  public static final String ARCH_PPC_32 = "powerpc";
+  public static final String ARCH_PPC_32_LE = "powerpcle";
+  public static final String ARCH_PPC_64 = "ppc64";
+  public static final String ARCH_PPC_64_LE = "ppc64le";
+  public static final String ARCH_X86_32 = "x86";
+  public static final String ARCH_X86_64 = "x86-64";
+
+  public static final Map<String, String> archMap = new HashMap<String, String>() {{
```

Comment:
Why is any of this public? Also I think both of these values (the ones from `os.arch` and the ones used internally) should have a specific term. Perhaps `architectureName` and `architectureType` respectively, considering that the "name" is more detailed and the "type" is more generic (more accurate would be to refer to it as "abi" but that may get too technical). This would become `architectureNameToTypeMap` or such. By the way, if you haven't noticed, I prefer longer names to abbreviations . 😛